app/Keycloak/Models/KeycloakClientModel.php

```diff
@@ -28,13 +29,15 @@ final class KeycloakClientModel extends UuidModel
 
     public function toDomain(): Client
     {
+        /** @var Config $config */
+        $config = App::get(Config::class);
+
         return new Client(
-            // Trying to use magic getters for eg $this->id gives a 0 back
             Uuid::fromString($this->id),
             Uuid::fromString($this->integration_id),
             Uuid::fromString($this->client_id),
             $this->client_secret,
-            RealmCollection::fromPublicName($this->realm)
+            $config->realms->fromPublicName($this->realm)
```

Comment:
I think the solutions from `Auth0Client` and `UiTiDv1Consumer` are more elegant. They use the enum `Auth0Tenant` and `UiTiDv1Environment` instead of a full domain object. When taking that approach you also have the added that you will not expose the key and secret of a realm to the UI

Reply:
I do not really see how I am exposing the key and secret to the ui, but I might not be aware of an Nova feature. Where could you see the secret of a realm in the ui?

I do not find this an an un-elegant design, the Realm is used a lot in other classes, for example the KeycloakApiClient to know to which realm to sent requests. 
If I remove it here I would have to add code in all these places to transform the Enum to a valid Realm object.

Reply:
In the context of the `KeycloakApiClient` you indeed need the `Realm` because you need to know the URL and the credentials. But in the context of a saved `KeycloackClient you don't need that extra information, you only need to know the environment.

This is also a hot topic in the context of DDD to pick meaningful boundaries or bounded contexts:
- make the boundary too small and you will miss information, so you need a lot of other domain objects to get enough information
- make the boundary too big and you will be dragging along unimportant information and have overhead in creating your domain objects